### PR TITLE
[ADD] product_bom_revision

### PR DESCRIPTION
**Pull request by emagdalenaC2i**
This module allows you to store the revision of a product and a BOM.

### CODE REVIEW

**Comment by emagdalenaC2i** [comment]
@Nikul-Chaudhary and @bodedra can you please review this PR?

File: product_revision/readme/USAGE.rst [hunk 1]
```diff
@@ -0,0 +1,12 @@
+* Go to Inventory > Master Data > Products
+* Create or edit a product.
+* Enter the firmware version and save.
```
Comment (bodedra) [inline product_revision/readme/USAGE.rst]:
```suggestion
* Enter the product revision and save.
```

File: product_revision/readme/USAGE.rst [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+* Check that the firmware version is saved and can be used to search.
```
Comment (bodedra) [inline product_revision/readme/USAGE.rst]:
```suggestion
* Check that the product revision is saved and can be used to search.
```

File: product_revision/README.rst [hunk 3]
```diff
@@ -0,0 +1,63 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+========================
+Product revision
+========================
+
+This module allows you to store the revision of a product and a BOM..
```
Comment (bodedra) [inline product_revision/README.rst]:
```suggestion
This module allows you to store the revision of a product and a BOM.
```

**Review by bodedra — APPROVED** [review]
Functional reviewed. :+1:  Added minor suggestion. Additionally remove Usage.rst (firmware revision) content and add following content.


* Go to Inventory > Master Data > Products
* Create or edit a product.
* Enter the Revision and save.
* Create or edit a BOM.
* Enter the Revision and save.

**Review by HviorForgeFlow — COMMENTED** [review]
I'm not used to revision workflows. I can't opine. Code is correct, missing some base_revision model to handle inheritances.

**Comment by emagdalenaC2i** [comment]
@HviorForgeFlow there are no revisions workflow just a new field 'revision' in products and bom

@patrickrwilson Please, can you review this PR?

**Comment by HviorForgeFlow** [comment]
Module name should explicitly clarify that this module brings BoM revision also.
mrp_bom_product_revision? or split into 2 modules.

**Comment by emagdalenaC2i** [comment]
I have rename it to product_bom_revision

**Review by patrickrwilson — APPROVED** [review]
LGTM

It might be useful to add tracking to the revision fields but not necessary.

**Review by nikul-serpentcs — APPROVED** [review]
Code Review LGTM :+1:

File: product_bom_revision/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Product BOM revision",
+    "summary": "Store the revision of a product and a BOM",
+    "version": "12.0.1.0.0",
+    "license": "AGPL-3",
+    "author": "C2i Change 2 improve,"
+              "Odoo Community Association (OCA)",
+    "category": "Stock",
```
Comment (rousseldenis) [inline product_bom_revision/__manifest__.py]: @emagdalenaC2i Could you change this to 'Warehouse' as it won't appear in the good category in OCA shop?

**Review by rousseldenis — APPROVED** [review]
Code review. Minor change.

I'm ok, but maybe split bom and product revision or make more generic module, because why these two (and not variant for instance)

File: product_bom_revision/__manifest__.py (hunk 4 above)
Comment (emagdalenaC2i) [reply product_bom_revision/__manifest__.py]: @rousseldenis ok, changed

**Comment by emagdalenaC2i** [comment]
@rousseldenis done

**Comment by rousseldenis** [comment]
@OCA/product-maintainers

**Comment by emagdalenaC2i** [comment]
@jgrandguillaume or @lasley  Please, could you check and merge the PR approved in this repository?

**Pull request merged**